sale_quick/models/sale_order.py

```diff
@@ -26,7 +27,8 @@ def _get_context_add_products(self):
         return res
 
     def _get_domain_add_products(self):
-        return []
+        field = self.env["sale.order.line"]._fields.get("product_id")
+        return safe_eval(field.domain, {"company_id": self.company_id.id})
```

Comment:
This is missing possible fields values, as domains may be "dynamic". Example: `"[('type_id', '=', type_id)]"`

Reply:
Thanks for the feedback, however, I'm not sure how to go about fixing it... What would you suggest?

Reply:
> This is missing possible fields values, as domains may be "dynamic". Example: "[('type_id', '=', type_id)]"

You're right.
however, I think that this case is quite theoritical and will not occures. and if it occures, it should be fixed by a custom glue module.
I mean, if a custom module add a domain like "[('type_id', '=', type_id)]" and if sale_quick is installed, the custom module (or another glue module) should update the function. _get_domain_add_products.
(with or without this patch).

For me, this implementation is OK.

I see it also robust because if the domain changes in the sale module, we have nothing to do in the OCA module.

Reply:
OK, I dismiss my request changes if you want to go ahead, but I don't think this solution is good enough. Maybe this can be handled at `fields_view_get` level.

Reply:
@pedrobaeza would you like for us to add a line in roadmap?